StreamChatClient_v3/APIClient/Endpoints/ChannelEndpoints.swift

```diff
@@ -0,0 +1,305 @@
+//
+// ChannelEndpoints.swift
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+extension Endpoint {
+    static func channels<ExtraData: ExtraDataTypes>(query: ChannelListQuery)
+        -> Endpoint<ChannelListEndpointPayload<ExtraData>> {
+        .init(path: "channels",
+              method: .get,
+              queryItems: [],
+              jsonQueryItems: ["payload": query],
+              body: nil)
+    }
+}
+
+struct ChannelListEndpointPayload<ExtraData: ExtraDataTypes>: Decodable {
+    /// A list of channels response (see `ChannelQuery`).
+    let channels: [ChannelEndpointPayload<ExtraData>]
+}
+
+struct ChannelEndpointPayload<ExtraData: ExtraDataTypes>: Decodable {
+    let channel: ChannelDetailPayload<ExtraData>
+    
+    let watcherCount: Int
+    
+    let members: [MemberEndpointPayload<ExtraData.User>]
+    
+    // TODO:
```

Comment:
Why don't we include these too?

Reply:
The same reason as for many other missing pieces: I've already spent 5 days on this and wanted to get it merged so others can work on it too. This is/was the last major blocker.